src/main/scala/firrtl/AddDescriptionNodes.scala

```diff
@@ -89,44 +143,97 @@ class AddDescriptionNodes extends Transform with PreservesAll[Transform] {
 
   override val dependents = Seq.empty
 
-  def onStmt(compMap: Map[String, Seq[String]])(stmt: Statement): Statement = {
-    stmt.map(onStmt(compMap)) match {
-      case d: IsDeclaration if compMap.contains(d.name) =>
-        DescribedStmt(DocString(StringLit.unescape(compMap(d.name).mkString("\n\n"))), d)
-      case other => other
+  def onStmt(compMap: Map[String, Seq[Description]])(stmt: Statement): Statement = {
+    val s = stmt.map(onStmt(compMap))
+    val sname = s match {
+      case d: IsDeclaration => Some(d.name)
+      case _ => None
+    }
+    val descs = sname.flatMap({ case name =>
+      compMap.get(name)
+    })
+    (descs, s) match {
+      case (Some(d), DescribedStmt(prevDescs, ss)) => DescribedStmt(prevDescs ++ d, ss)
+      case (Some(d), ss) => DescribedStmt(d, ss)
+      case (None, _) => s
     }
   }
 
-  def onModule(modMap: Map[String, Seq[String]], compMaps: Map[String, Map[String, Seq[String]]])
+  def onModule(modMap: Map[String, Seq[Description]], compMaps: Map[String, Map[String, Seq[Description]]])
     (mod: DefModule): DefModule = {
-    val (newMod, portDesc: Map[String, Description]) = compMaps.get(mod.name) match {
-      case None => (mod, Map.empty)
-      case Some(compMap) => (mod.mapStmt(onStmt(compMap)), mod.ports.collect {
-        case p @ Port(_, name, _, _) if compMap.contains(name) =>
-          name -> DocString(StringLit.unescape(compMap(name).mkString("\n\n")))
-      }.toMap)
-    }
+    val compMap = compMaps.getOrElse(mod.name, Map())
+    val newMod = mod.mapStmt(onStmt(compMap))
+    val portDesc = mod.ports.collect {
+      case p @ Port(_, name, _, _) if compMap.contains(name) =>
+        name -> compMap(name)
+    }.toMap
```

Comment:
I'm not convinced this is actually better, kind of fun though 😜 
```suggestion
    val portDesc = mod.ports.flatMap { case Port(_, name, _, _) => compMap.get(name).map(name -> _) }.toMap
```